trace-to-events/src/main.rs

```diff
@@ -101,32 +115,43 @@ async fn main() {
                             .inc();
                         match root_as_digitizer_analog_trace_message(payload) {
                             Ok(thing) => {
-                                match producer
-                                    .send(
+                                let mut fbb = FlatBufferBuilder::new();
+                                processing::process(
+                                    &mut fbb,
+                                    &thing,
+                                    &DetectorSettings {
+                                        polarity: &args.polarity,
+                                        baseline: args.baseline,
+                                        mode: &args.mode,
+                                    },
+                                    args.save_file.as_deref(),
+                                );
+
+                                let future = producer
+                                    .send_result(
                                         FutureRecord::to(&args.event_topic)
-                                            .payload(&processing::process(
-                                                &thing,
-                                                &args.mode,
-                                                args.save_file.as_deref(),
-                                            ))
+                                            .payload(fbb.finished_data())
                                             .key("test"),
-                                        Duration::from_secs(0),
                                     )
-                                    .await
-                                {
-                                    Ok(_) => {
-                                        trace!("Published event message");
-                                        metrics::MESSAGES_PROCESSED.inc();
+                                    .expect("Producer sends");
+
+                                tokio::spawn(async {
```

Comment:
Does this actually help throughput? I would have expected the benefit of handling only the transmission of the event list message would not really be of much benefit when the detection and de/serialisation all happens sequentially.

To be clear, since event detection is already a trivially parallisable problem at the digitiser level, I am of the opinion that adding complexity to the event detector is not worth effort when it can be horizontally scaled with no changes required.